<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:include filename="base.urdf.xacro"/>
  <xacro:include filename="left_wheel.urdf.xacro"/>
  <xacro:include filename="right_wheel.urdf.xacro"/>
  <xacro:include filename="left_caster.urdf.xacro"/>
  <xacro:include filename="right_caster.urdf.xacro"/>
  <xacro:include filename="mid360.urdf.xacro"/>

  <!-- <link name="world"> -->
  <!-- </link> -->
  <!---->
  <!-- <joint name="world_joint" type="fixed"> -->
  <!--   <parent link="world"/> -->
  <!--   <child link="base_link"/> -->
  <!--   <origin xyz="0 0 0.01"/> -->
  <!-- </joint> -->

  <link name="base_link"/>

  <xacro:base name="base" parent="base_link">
    <origin xyz="0 0 0.38" rpy="0 0 0"/> <!-- wheels touch the ground with z=0.3784-->
  </xacro:base>

  <joint name="odom_joint" type="fixed">
    <parent link="base"/>
    <child link="odom"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>

  <link name="odom"/>

  <xacro:left_wheel name="left_wheel" parent="base">
    <origin xyz="-0.3129 0.0 -0.2234" rpy="0 0 0"/> 
  </xacro:left_wheel>

  <xacro:right_wheel name="right_wheel" parent="base">
    <origin xyz="0.3129 0.0 -0.2234" rpy="0 0 0"/> 
  </xacro:right_wheel>

  <xacro:left_caster name="left_caster" parent="base">
    <origin xyz="-0.3006 0.4594 -0.3152" rpy="0 0 0"/> 
  </xacro:left_caster>

  <xacro:right_caster name="right_caster" parent="base">
    <origin xyz="0.2951 0.471064 -0.3102" rpy="0 0 0"/> 
  </xacro:right_caster>

  <xacro:mid360 name="livox_frame" parent="base">
    <origin xyz="-0.002 -0.0632 0.9655" rpy="0 0 0"/> 
  </xacro:mid360>

</robot>
